### ISSUE TITLE
Adding a file-type to ParserExtensions forces C++ SyntaxHighlighting

### ISSUE THREAD

**Issue by domme** [slynch8/10x#1837]
In our project we're using custom file-extensions for our (own but hlsl-like) shader code files. We've created a custom SyntaxHighlighting file for those extensions, which work great. However, we'd **also** like to benefit from parsing those files as if they were C++ (to e.g. get auto-completion and basic function-parsing support) so we've also added those extensions to the `ParserExtensions` list in the `10x_settings`-file. After doing a re-parse on the whole solution however, the custom syntax highlighting is gone while editing a file (notably, when leaving the focus - e.g. by switching over to the other pane in a dual-pane layout - the custom SyntaxHighligting reappears - until focus goes back to that pane).

**Comment by slynch8** [comment]
The parser coloring will replace the syntax highlighting coloring (the syntax highlighting is only really for non-C++ files) so you can't have both unfortunately. But you should get coloring as if it was C++ code.

**Comment by domme** [comment]
Ah okay, so this is by design then and not a bug? 
I had assumed forcing the C++ syntax highlighting in such a case must be unintentional since the default 10x installation also includes a `shader.10x_syntax` file for extensions like .hlsl or .glsl but also lists those in the `ParserExtensions` list with default settings - which would basically make the existance of the syntax-file pointless?
Anyway, thanks for getting back to me. C++-like syntax coloring on these files works fine as you said :)

**Comment by slynch8** [comment]
I've created a new bug for this
https://github.com/slynch8/10x/issues/1853

**Comment by domme** [comment]
Hey @slynch8 - Just noticed that this issue is closed since some time now, thanks a lot!
However, when trying it out just now (version 1.0.168) I'm still seeing the exact same behavior I've reported initially:

I have a custom shader file extension ".mrpr" for which I added an "mrpr.10x_syntax" file, which e.g. contains this section to add syntax highlighting for "float4" types: 
`Language.Keyword:`
`WORD(float4)`

I've confirmed that this works perfectly fine as long as I'm **not** adding the ".mrpr" extension to the `ParserExtensions: ` list in the 10x_settings file. As soon as I do that, the syntax-highlighting from the "mrpr.10x_syntax" file is ignored and default c++-coloring is used instead. Is there anything I'm doing wrong here? The closed bug #1853 mentions that only "non-language" settings are overridden? `Language.Keyword:` certainly sounds like a langauge-feature though, so maybe I'm using a wrong way to define my syntax coloring?
